File: specs/master/query_dsl/compound/bool/bool_query.ts

/**namespace:QueryDsl.Compound.Bool */
/**custom_serialization*/
interface bool_query {
	must: query_container[];
	must_not: query_container[];
	should: query_container[];
	filter: query_container[];
	minimum_should_match: minimum_should_match;
	disable_coord: boolean;
	Locked: boolean;
}

File: output/csharp/QueryDsl/Compound/Bool/BoolQuery.cs
using Nest.QueryDsl;
using Nest.CommonOptions;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.QueryDsl {

	public class BoolQuery  {
		
		[DataMember(Name="filter")]
		public List<QueryContainer> Filter { get; set; }


		[DataMember(Name="locked")]
		public bool Locked { get; set; }


		[DataMember(Name="minimum_should_match")]
		public MinimumShouldMatch MinimumShouldMatch { get; set; }


		[DataMember(Name="must")]
		public List<QueryContainer> Must { get; set; }


		[DataMember(Name="must_not")]
		public List<QueryContainer> MustNot { get; set; }


		[DataMember(Name="should")]
		public List<QueryContainer> Should { get; set; }

	}
}
